GW2EIBuilders/HTMLBuilder.cs

```diff
@@ -1,308 +1,338 @@
-﻿using System;
-using System.IO;
-using System.IO.Compression;
+﻿using System.IO.Compression;
 using System.Text;
 using GW2EIBuilders.HtmlModels;
 using GW2EIEvtcParser;
-using Newtonsoft.Json;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Tracing;
+using GW2EIEvtcParser.EIData;
 
 [assembly: System.CLSCompliant(false)]
-namespace GW2EIBuilders
+namespace GW2EIBuilders;
+
+class IntTupleArrayConverter : JsonConverter<(int, int)>
 {
-    public class HTMLBuilder
+    public override (int, int) Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
     {
-        private static readonly UTF8Encoding NoBOMEncodingUTF8 = new UTF8Encoding(false);
+        throw new NotImplementedException();
+    }
 
-        private readonly string _eiJS;
-        private readonly string _eiCRJS;
-        private readonly string _eiHealingExtJS;
+    public override void Write(Utf8JsonWriter writer, (int, int) value, JsonSerializerOptions options)
+    {
+        writer.WriteStartArray();
+        writer.WriteNumberValue(value.Item1);
+        writer.WriteNumberValue(value.Item2);
+        writer.WriteEndArray();
+    }
+}
+
+// compile-time generated serialization logic
+[JsonSourceGenerationOptions(GenerationMode = JsonSourceGenerationMode.Metadata | JsonSourceGenerationMode.Serialization,
+    PropertyNamingPolicy = JsonKnownNamingPolicy.CamelCase,
+    IncludeFields = true, WriteIndented = false, NumberHandling = JsonNumberHandling.AllowNamedFloatingPointLiterals,
+    Converters = [ typeof(IntTupleArrayConverter) ]
+)]
+[JsonSerializable(typeof(LogDataDto))]
+[JsonSerializable(typeof(AgentConnector.AgentConnectorDescriptor))]
```

Comment:
I'm guessing we would not need those AgentConnector serializables if the GetConnected to API were to return a generic ConnectorDescriptor class with the others declared as JsonDerived? Just as a note to myself, trying to get used to System.Text.Json useage.

Reply:
Yes. In the process of clearing up the visibility question im about to add the following comment to that code :D :

```cs
[JsonSerializable(typeof(LogDataDto))]
//NOTE(Rennorb): The following types need to be manually added to the generator, because they are used as opaque object fields.
// The generator cannot know the types of those fields, and we want to avoid having to use slow reflection to find out the members of those arbitrary types.
// Specifying them here tells the generator to generate switches for these specific types whenever an opaque Object needs to be serialized.
[JsonSerializable(typeof(AgentConnector.AgentConnectorDescriptor))]
[JsonSerializable(typeof(PositionConnector.PositionConnectorDescriptor))]
[JsonSerializable(typeof(AgentFacingConnector.AgentFacingConnectorDescriptor))]
...
```

Reply:
I should also mention that this is not required to use the System.Text.Json serializer, only if you use the generated (de)serializers. These are quite a bit faster because they don't use reflection to find fields, but you need to specify more data in cases that use inheritance (which using "object" as the base is just a special case of).